packages/core/src/useForm/useFormActions.ts

```diff
@@ -170,32 +191,142 @@ export function useFormActions<TForm extends FormObject = FormObject, TOutput ex
+    pathOrStateOrUndefined?: TPath | Partial<ResetState<TForm>>,
+    stateOrOptsOrUndefined?: Partial<ResetState<PathValue<TForm, TPath>>> | SetValueOptions,
+    optsOrUndefined?: SetValueOptions,
+  ): Promise<void> {
+    /**
+     * CASE 1: reset()
+     * Reset the form to its initial state.
+     */
+    if (pathOrStateOrUndefined === undefined) {
+      wasSubmitted.value = false;
+
+      form.revertValues();
+      form.revertTouched();
+      form.revertDirty();
+      submitAttemptsCount.value = 0;
+      isSubmitAttempted.value = false;
+
+      form.clearErrors();
+      form.clearSubmitErrors();
+
+      await validate();
 
-    if (state?.touched) {
-      form.setInitialTouched(state.touched, opts);
+      return Promise.resolve();
     }
 
-    wasSubmitted.value = false;
+    /**
+     * CASE 2: reset(state, opts?)
+     * Reset the form to a specific state.
+     */
+    if (isResetState<TForm>(pathOrStateOrUndefined)) {
+      const state = pathOrStateOrUndefined;
+      const opts = stateOrOptsOrUndefined as SetValueOptions;
 
-    form.revertValues();
-    form.revertTouched();
-    form.revertDirty();
-    submitAttemptsCount.value = 0;
-    isSubmitAttempted.value = false;
+      if (state.value) {
+        form.setInitialValues(state.value);
+      }
 
-    if (state?.revalidate ?? true) {
-      await validate();
-      return;
+      if (state.touched) {
+        // Add type assertion here to help TypeScript understand the structure
+        form.setInitialTouched(state.touched as Partial<TouchedSchema<TForm>>, opts);
+      }
+
+      wasSubmitted.value = false;
+      submitAttemptsCount.value = 0;
+      isSubmitAttempted.value = false;
+
+      form.revertValues();
+      form.revertTouched();
+      form.revertDirty();
+
+      if (state.revalidate ?? true) {
+        await validate();
+        return;
+      }
+
+      form.clearErrors();
+      form.clearSubmitErrors();
+
+      return Promise.resolve();
     }
 
-    form.clearErrors();
-    form.clearSubmitErrors();
+    /**
+     * Case 3 and 4.
+     *
+     * Extract the path from the first argument.
+     */
+    const path = pathOrStateOrUndefined;
+
+    /**
+     * CASE 3: reset(path)
+     * Reset a specific part of the form.
+     */
+    if (stateOrOptsOrUndefined === undefined) {
+      form.revertValues(path);
+      form.revertTouched(path);
+      form.revertDirty(path);
+      form.clearErrors(path);
+
+      return Promise.resolve();
+    }
+
+    /**
+     * CASE 4: reset(path, state, opts?)
+     * Reset a specific part of the form to a specific state.
+     */
+    if (isResetState<PathValue<TForm, TPath>>(stateOrOptsOrUndefined)) {
+      const state = stateOrOptsOrUndefined;
+      const opts = optsOrUndefined;
 
-    return Promise.resolve();
+      if (state.value) {
+        form.setInitialValuesPath(path, state.value, opts);
+      }
+
+      if (state.touched) {
+        form.updateTouchedPath(path, state.touched as any, opts);
```

Comment:
Is there a better way to pass this without casting to `any`?

Reply:
Kinda why I didn't want to type these in the first place, its a long chain of types that needs to be maintained, but that's me being lazy.

I will see what I can do.